app/lib/identity_tijuana/member_sync.rb

```diff
@@ -6,25 +6,55 @@ def self.export_member(member, sync_id)
       begin
         return if member.ghosting_started?
 
-        ext_id = MemberExternalId.find_by(system: 'tijuana', member: member)
-        user_id = ext_id.external_id if ext_id.present?
-        if user_id.present?
-          user = User.find_by(id: user_id)
-          ext_id.destroy! if user.blank?
+        ext_ids = MemberExternalId.where(system: 'tijuana', member: member).to_a
+        # Destroy any dangling external ID references, remove any
+        # external ids from the list if not found
+        ext_ids.select do |ext_id|
+          user = User.find_by(id: ext_id.external_id.to_i)
+          if user.blank?
+            Rails.logger.info(
+              "[IdentityTijuana::export_member]: " \
+              "Removing dangling external id for Id member #{member.id}, " \
+              "TJ user #{ext_id.external_id} " \
+              "(sync_id=#{sync_id}, sync_type=#{sync_type}, " \
+              "sync_direction=export_member)"
+            )
+            ext_id.destroy!
+          end
+          user.present?
         end
-        if user.blank?
-          sync_type = :merge
-          email = member.email
-          user = User.find_by(email: email) if email.present?
-          sync_type = :create if user.blank?
+        if ext_ids.count == 0
+          # XXX this will fail for de-normalised and/or cleaned emails
+          user = User.find_by(email: member.email) if member.email.present?
+
+          if user.nil?
+            sync_type = :create
+          else
+            sync_type = :merge
```

Comment:
`export_member ` and `import_user` seem to share some same logic steps, maybe it would good to abstract common logic into a reusable helper method?

Reply:
Yeah I was having the exact same thought this morning - was thinking about pushing that down into the sync method.

But then again, while there is some common structure there, the logic around it is pretty different, and some required special case handling that is very different. Since it's also not a /lot/ of duplication, for now I think it should stay as-is.